pkg/model/field.go

```diff
@@ -88,36 +89,76 @@ type Field struct {
 // "// This field contains the identifier for the cluster
 //
 //	// running the cache services
-//	//
 //	// please note that this field is updated on the service
 //	// side"
 func (f *Field) GetDocumentation() string {
+	cfg := f.GetFieldDocsConfig()
+
 	hasShapeDoc := false
-	var sb strings.Builder
+	var prepend strings.Builder
+	var out strings.Builder
+
 	if f.ShapeRef != nil {
 		if f.ShapeRef.Documentation != "" {
 			hasShapeDoc = true
-			sb.WriteString(f.ShapeRef.Documentation)
+			out.WriteString(f.ShapeRef.Documentation)
 		}
 	}
-	if f.FieldConfig != nil {
-		if f.FieldConfig.Documentation != nil {
-			if hasShapeDoc {
-				sb.WriteString("\n//\n")
-			}
-			// Strip any leading comment slashes from the config option
-			// docstring since we'll be automatically adding the Go comment
-			// slashes to the beginning of each new line
-			cfgDoc := *f.FieldConfig.Documentation
-			lines := strings.Split(cfgDoc, "\n")
-			numLines := len(lines)
-			for x, line := range lines {
-				sb.WriteString("// ")
-				sb.WriteString(strings.TrimLeft(line, "/ "))
-				if x < (numLines - 1) {
-					sb.WriteString("\n")
-				}
-			}
+	if cfg == nil {
+		return out.String()
+	}
+
+	// Ensure configuration has exclusive options
+	if cfg.Override != nil && (cfg.Append != nil || cfg.Prepend != nil) {
+		panic("Documentation cannot contain override and prepend/append. Field: " + f.CRD.Kind)
+	}
+
+	if cfg.Override != nil {
+		return f.formatUserProvidedDocstring(*cfg.Override)
+	}
+
+	if cfg.Append != nil {
+		if hasShapeDoc {
+			out.WriteString("\n//\n")
+		}
+		out.WriteString(f.formatUserProvidedDocstring(*cfg.Append))
+	}
+
+	if cfg.Prepend != nil {
+		prepend.WriteString(f.formatUserProvidedDocstring(*cfg.Prepend))
+		if hasShapeDoc {
+			prepend.WriteString("\n//\n")
+		}
+	}
+
+	return prepend.String() + out.String()
+}
+
+// formatUserProvidedDocstring sanitises a doc string provided by the user so
+// that it fits the format of the existing AWS SDK GoDocs. This method will
+// split it by lines, strip it of leading whitespace and slashes and prepend
+// `//` to each line.
+func (f *Field) formatUserProvidedDocstring(in string) string {
+	var sb strings.Builder
+	// Strip any leading comment slashes from the config option
+	// docstring since we'll be automatically adding the Go comment
+	// slashes to the beginning of each new line
+	lines := strings.Split(in, "\n")
+
+	// Traverse in reverse order until, deleting empty lines
```

Comment:
Any reason you wanted to delete empty newlines? Sometimes a long docstring is better broken into paragraphs.

Reply:
Only deleting trailing newlines here